src/libasr/codegen/x86_assembler.h

```diff
@@ -1195,6 +1237,110 @@ class X86Assembler {
             X86Reg::ecx, base, index, scale, disp, true);
         EMIT("fimul dword " + m2s(base, index, scale, disp));
     }
+
+    // Move or Merge Scalar Double Precision Floating-Point Value
+    void asm_movsd_r64_m64(X64FReg r64, X64Reg *base, X64Reg *index,
+                uint8_t scale, int64_t disp) {
+        X86Reg r32 = X86Reg(r64 & 7);
+        X86Reg* base32 = nullptr, *index32 = nullptr;
+        if (base) {
+            base32 = new X86Reg;
+            *base32 = X86Reg(*base & 7);
+        }
+        if (index) {
+            index32 = new X86Reg;
+            *index32 = X86Reg(*index & 7);
+        }
+        m_code.push_back(m_al, 0xf2);
+        m_code.push_back(m_al, rex(1, r64 >> 3, (index32 ? (*index32 >> 3) : 0), (base32 ? (*base32 >> 3) : 0)));
+        m_code.push_back(m_al, 0x0f);
+        m_code.push_back(m_al, 0x10);
+        modrm_sib_disp(m_code, m_al,
+                    r32, base32, index32, scale, (int32_t)disp, true);
+        EMIT("movsd " + r2s(r64) + ", " + m2s(base, index, scale, disp));
+    }
+
+    // Move or Merge Scalar Double Precision Floating-Point Value
+    void asm_movsd_m64_r64(X64Reg *base, X64Reg *index,
+                uint8_t scale, int64_t disp, X64FReg r64) {
+        X86Reg r32 = X86Reg(r64 & 7);
+        X86Reg* base32 = nullptr, *index32 = nullptr;
+        if (base) {
+            base32 = new X86Reg;
+            *base32 = X86Reg(*base & 7);
```

Comment:
This seems to be leaking memory. If you need to allocate, use the Allocator.

However, we should not need to allocate these simple structures, just use them by value. You can let the caller provide the allocation using a local variable.

Reply:
I fixed it in [X86Assembler: Use local varaibles for index32 and base32](https://github.com/lcompilers/lpython/pull/1402/commits/cf2c202aeb01c2a476885542260fb992798aa350).